File: com/RaghavStaadExtractor/AssignSectionDatabase.vb
Imports Microsoft.Office.Interop.Excel
Imports System.IO
Imports System.Runtime.InteropServices

<ComVisible(True)>
<Guid("B3E62114-9A8A-4A61-B71F-3217288DB2E2")>
<ProgId("RaghavStaadExtractor.AssignSectionDatabase")>
Public Class AssignSectionDatabase

    Public Sub CopySectionDatabaseToSheet3()
        Dim excelApp As Application = Nothing
        Dim sourceWB As Workbook = Nothing
        Dim sourceSheet As Worksheet = Nothing
        Dim targetWB As Workbook = Nothing
        Dim targetSheet As Worksheet = Nothing

        Try
            excelApp = Marshal.GetActiveObject("Excel.Application")

            excelApp.ScreenUpdating = False

            targetWB = excelApp.ActiveWorkbook
            targetSheet = CType(targetWB.Sheets(3), Worksheet)
            targetSheet.Cells.Clear()

            Dim folderPath As String = targetWB.Path & "\"
            Dim fileName As String = "RAGHAV DATABASE.xlsm"
            Dim filePath As String = Path.Combine(folderPath, fileName)

            If Not File.Exists(filePath) Then
                MsgBox("File '" & fileName & "' not found in the current folder.", MsgBoxStyle.Exclamation)
                Exit Sub
            End If

            sourceWB = excelApp.Workbooks.Open(filePath, [ReadOnly]:=True)
            sourceSheet = CType(sourceWB.Sheets(1), Worksheet)

            Dim lastRow As Long = sourceSheet.Cells(sourceSheet.Rows.Count, 1).End(XlDirection.xlUp).Row
            Dim lastCol As Long = sourceSheet.Cells(5, sourceSheet.Columns.Count).End(XlDirection.xlToLeft).Column
            Dim dataRange As Range = sourceSheet.Range(sourceSheet.Cells(5, 1), sourceSheet.Cells(lastRow, lastCol))

            Dim destRange As Range = targetSheet.Range("A1").Resize(dataRange.Rows.Count, dataRange.Columns.Count)
            destRange.Value = dataRange.Value
            destRange.Font.Color = RGB(0, 0, 139)

            sourceWB.Close(False)

            MsgBox("Staad Section Database, Perfectly Aligned!", MsgBoxStyle.Information)

        Catch ex As Exception
            MsgBox("Error: " & ex.Message, MsgBoxStyle.Critical)

        Finally
            If sourceWB IsNot Nothing Then Marshal.ReleaseComObject(sourceWB)
            If sourceSheet IsNot Nothing Then Marshal.ReleaseComObject(sourceSheet)
            If targetSheet IsNot Nothing Then Marshal.ReleaseComObject(targetSheet)
            If targetWB IsNot Nothing Then Marshal.ReleaseComObject(targetWB)
            If excelApp IsNot Nothing Then excelApp.ScreenUpdating = True

            sourceSheet = Nothing
            sourceWB = Nothing
            targetSheet = Nothing
            targetWB = Nothing
            excelApp = Nothing

            GC.Collect()
            GC.WaitForPendingFinalizers()
        End Try

    End Sub

End Class

File: com/RaghavStaadExtractor_decompiled_fresh/AssignSectionDatabase.cs
// ===== RaghavStaadExtractor.AssignSectionDatabase =====

// Decompiled member-by-member from RaghavStaadExtractor.dll



public AssignSectionDatabase()
{
}


using System;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using Microsoft.Office.Interop.Excel;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

public void CopySectionDatabaseToSheet3()
{
	Application application = null;
	Workbook workbook = null;
	Worksheet worksheet = null;
	Workbook workbook2 = null;
	Worksheet worksheet2 = null;
	try
	{
		application = (Application)Marshal.GetActiveObject("Excel.Application");
		application.ScreenUpdating = false;
		workbook2 = application.ActiveWorkbook;
		worksheet2 = (Worksheet)workbook2.Sheets[3];
		worksheet2.Cells.Clear();
		string path = workbook2.Path + "\\";
		string text = "RAGHAV DATABASE.xlsm";
		string text2 = Path.Combine(path, text);
		if (!File.Exists(text2))
		{
			Interaction.MsgBox("File '" + text + "' not found in the current folder.", MsgBoxStyle.Exclamation);
			return;
		}
		workbook = application.Workbooks.Open(text2, RuntimeHelpers.GetObjectValue(Missing.Value), true, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value));
		worksheet = (Worksheet)workbook.Sheets[1];
		long num = Conversions.ToLong(NewLateBinding.LateGet(NewLateBinding.LateGet(worksheet.Cells[worksheet.Rows.Count, 1], null, "End", new object[1] { XlDirection.xlUp }, null, null, null), null, "Row", new object[0], null, null, null));
		long num2 = Conversions.ToLong(NewLateBinding.LateGet(NewLateBinding.LateGet(worksheet.Cells[5, worksheet.Columns.Count], null, "End", new object[1] { XlDirection.xlToLeft }, null, null, null), null, "Column", new object[0], null, null, null));
		Range range = ((_Worksheet)worksheet).get_Range(RuntimeHelpers.GetObjectValue(worksheet.Cells[5, 1]), RuntimeHelpers.GetObjectValue(worksheet.Cells[num, num2]));
		Range range2 = ((_Worksheet)worksheet2).get_Range((object)"A1", RuntimeHelpers.GetObjectValue(Missing.Value)).get_Resize((object)range.Rows.Count, (object)range.Columns.Count);
		range2.set_Value(RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(range.get_Value(RuntimeHelpers.GetObjectValue(Missing.Value))));
		range2.Font.Color = Information.RGB(0, 0, 139);
		workbook.Close(false, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value));
		Interaction.MsgBox("Staad Section Database, Perfectly Aligned!", MsgBoxStyle.Information);
	}
	catch (Exception ex)
	{
		ProjectData.SetProjectError(ex);
		Exception ex2 = ex;
		Interaction.MsgBox("Error: " + ex2.Message, MsgBoxStyle.Critical);
		ProjectData.ClearProjectError();
	}
	finally
	{
		if (workbook != null)
		{
			Marshal.ReleaseComObject(workbook);
		}
		if (worksheet != null)
		{
			Marshal.ReleaseComObject(worksheet);
		}
		if (worksheet2 != null)
		{
			Marshal.ReleaseComObject(worksheet2);
		}
		if (workbook2 != null)
		{
			Marshal.ReleaseComObject(workbook2);
		}
		if (application != null)
		{
			application.ScreenUpdating = true;
		}
		worksheet = null;
		workbook = null;
		worksheet2 = null;
		workbook2 = null;
		application = null;
		GC.Collect();
		GC.WaitForPendingFinalizers();
	}
}
